runtime/src/main/scala/akka/grpc/GrpcProtocol.scala

```diff
@@ -121,7 +119,13 @@ object GrpcProtocol {
    * Detects which gRPC protocol variant is indicated by a mediatype.
    * @return a [[GrpcProtocol]] matching the request mediatype if known.
    */
-  def detect(mediaType: jmodel.MediaType): Option[GrpcProtocol] = protocols.find(_.mediaTypes.contains(mediaType))
+  def detect(mediaType: jmodel.MediaType): Option[GrpcProtocol] = mediaType.subType match {
+    // FIXME: do we need to check mainType?
```

Comment:
it's not very likely that we would encounter other types where the subtype exactly matches the constants we know of here, so just matching the subtype might indeed be OK

Reply:
```suggestion
    // mainType is not checked. We assume it's the right one.
```